modules/gke/gsas.tf

```diff
@@ -32,4 +32,6 @@ resource "google_project_iam_member" "gke_admin" {
 # https://registry.terraform.io/providers/hashicorp/google/latest/docs/resources/google_service_account_key#private_key
 resource "google_service_account_key" "gke_cluster_access_key" {
   service_account_id = google_service_account.gke_cluster_access.name
+
+  depends_on = [google_project_iam_member.gke_admin]
```

Comment:
This is needed to ensure the service account roles aren't destroy when the keys are still used as otherwise cleanup operations might fail.